examples/diabetes_example/assets/functions/federated_analytics_functions.py

```diff
@@ -0,0 +1,73 @@
+import json
+from collections import defaultdict
+
+import pickle
+
+import numpy as np
+import pandas as pd
+import substratools as tools
+
+
+@tools.register
+def local_step1(inputs, outputs, task_properties):
+    df = inputs["datasamples"]
+    states = {
+        "n_samples": len(df),
+        "means": df.select_dtypes(include=np.number).sum().to_dict(),
+        "counts": {name: series.value_counts().to_dict() for name, series in df.select_dtypes(include='category').items()}
+    }
+    save_states(states, outputs["local_analytics_first_moments"])
+
+
+@tools.register
+def local_step2(inputs, outputs, task_properties):
+    df = inputs["datasamples"]
+    shared_states = load_states(inputs["shared_states"])
+    means = pd.Series(shared_states["means"])
+    states = {
+        "n_samples": len(df),
+        "std": np.power(df.select_dtypes(include=np.number) - means, 2).sum(),
+    }
+    save_states(states, outputs["local_analytics_second_moments"])
+
+
+@tools.register
+def aggregate(inputs, outputs, task_properties):
+    shared_states = [load_states(path) for path in inputs["local_analytics_list"]]
+
+    total_len = 0
+    for state in shared_states:
+        total_len += state["n_samples"]
```

Comment:
```suggestion
        total_len += state.pop("n_samples")
```

Reply:
I'm not sure I want to remove the value from my `state` dict. Plus I try to avoid using side-effects if I can.